File: python/30d_1367B_EvenArray.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        even_out_of_place, odd_out_of_place = 0, 0
        n = int(input())
        array = input().split()
        for i in range(n):
            array[i] = int(array[i])
            if i % 2 == 0:
                if array[i] % 2 != 0:
                    even_out_of_place += 1
            else:
                if array[i] % 2 != 1:
                    odd_out_of_place += 1
        if even_out_of_place != odd_out_of_place:
            ans.append(-1)
        else:
            ans.append(even_out_of_place)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/30d_1367B_EvenArray.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, a, no_of_0, no_of_1, even_out_of_place, odd_out_of_place;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        no_of_0 = 0, no_of_1 = 0, even_out_of_place = 0, odd_out_of_place = 0;
        std::cin>>n;
        for(int j = 0; j < n; j++){
            std::cin>>a;
            if(a % 2 == 0){
                no_of_0++;       
                if((a % 2) != (j % 2)){
                    even_out_of_place++;
                }
            }
            else{
                no_of_1++;       
                if((a % 2) != (j % 2)){
                    odd_out_of_place++;
                }
            }
        }
        if (n % 2 == 0){
            if(no_of_0 != no_of_1){
                ans.push_back(-1);
            }
            else{
                if(even_out_of_place != odd_out_of_place){
                    ans.push_back(-1);
                }
                else{
                    ans.push_back(even_out_of_place);
                }
            }
        }
        else{
            if(no_of_0 < no_of_1){
                ans.push_back(-1);
            }
            else{
                if(even_out_of_place != odd_out_of_place){
                    ans.push_back(-1);
                }
                else{
                    ans.push_back(even_out_of_place);
                }
            }
        }
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}